helix-term/src/application.rs

```diff
@@ -735,9 +743,11 @@ impl Application {
                                 doc.text(),
                                 language_id,
                             ));
+
+                            helix_event::dispatch(helix_view::events::DocumentDidOpen { doc });
```

Comment:
Ideally we should only be dispatching the internal DocumentDidOpen when we initially open a buffer. Here instead of using `DocumentDidOpen` we should be able to use `pull_diagnostics_for_document`

Reply:
Changed to use `pull_diagnostics_for_document` directly